hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java

```diff
@@ -71,124 +72,130 @@ public class ChunkKeyHandler extends KeyHandler implements
+            LOG.error("Cant execute getBlock on this node");
+            continue;
+          }
+          tempchunks = entry.getValue().getBlockData().getChunksList();
+          ContainerProtos.ContainerDataProto containerData =
+              containerOperationClient.readContainer(keyLocation
+                  .getContainerID(), pipeline);
+          for (ContainerProtos.ChunkInfo chunkInfo : tempchunks) {
+            String fileName = containerLayoutVersion.getChunkFile(new File(
+                    getChunkLocationPath(containerData.getContainerPath())),
+                keyLocation.getBlockID(),
+                ChunkInfo.getFromProtoBuf(chunkInfo)).toString();
+            chunkPaths.add(fileName);
+            ChunkDetails chunkDetails = new ChunkDetails();
+            chunkDetails.setChunkName(fileName);
+            chunkDetails.setChunkOffset(chunkInfo.getOffset());
+            chunkDetailsList.add(chunkDetails);
+          }
+          containerChunkInfoVerbose.setContainerPath(containerData
+              .getContainerPath());
+          containerChunkInfoVerbose.setPipeline(keyLocation.getPipeline());
+          containerChunkInfoVerbose.setChunkInfos(chunkDetailsList);
+          containerChunkInfo.setFiles(chunkPaths);
+          containerChunkInfo.setPipelineID(
+              keyLocation.getPipeline().getId().getId());
+          Gson gson = new GsonBuilder().create();
+          if (isVerbose()) {
+            element = gson.toJsonTree(containerChunkInfoVerbose);
+          } else {
+            element = gson.toJsonTree(containerChunkInfo);
+          }
+          jsonObj.addProperty("Datanode-HostName", entry.getKey()
+              .getHostName());
+          jsonObj.addProperty("Datanode-IP", entry.getKey()
+              .getIpAddress());
+          jsonObj.addProperty("Container-ID", containerId);
+          jsonObj.addProperty("Block-ID", keyLocation.getLocalID());
+          jsonObj.add("Locations", element);
+          responseFromAllNodes.add(jsonObj);
         }
-        jsonObj.addProperty("Datanode-HostName", entry.getKey().getHostName());
-        jsonObj.addProperty("Datanode-IP", entry.getKey().getIpAddress());
-        jsonObj.addProperty("Container-ID", containerId);
-        jsonObj.addProperty("Block-ID", keyLocation.getLocalID());
-        jsonObj.add("Locations", element);
-        responseFromAllNodes.add(jsonObj);
+        responseArrayList.add(responseFromAllNodes);
         xceiverClientManager.releaseClientForReadData(xceiverClient, false);
+        xceiverClient = null;
       }
-      responseArrayList.add(responseFromAllNodes);
+      result.add("KeyLocations", responseArrayList);
+      Gson gson2 = new GsonBuilder().setPrettyPrinting().create();
+      String prettyJson = gson2.toJson(result);
+      System.out.println(prettyJson);
+    } finally {
+      xceiverClientManager.close();
+      containerOperationClient.close();
+      ozoneManagerClient.close();
```

Comment:
`xceiverClientManager` is closed in `ContainerOperationClient#close`, and I don't think we usually close the `OzoneManagerClient` obtained from `RpcClient`.

Reply:
Done.